optuna/visualization/_slice.py

```diff
@@ -21,6 +24,82 @@
 _logger = get_logger(__name__)
 
 
+class _SliceSubplotInfo(NamedTuple):
+    param_name: str
+    x: List[Any]
+    y: List[float]
+    trial_numbers: List[int]
+    is_log: bool
+    is_numerical: bool
+
+
+class _SlicePlotInfo(NamedTuple):
+    target_name: str
+    subplots: List[_SliceSubplotInfo]
+
+
+def _get_slice_subplot_info(
+    trials: List[FrozenTrial],
+    param: str,
+    target: Optional[Callable[[FrozenTrial], float]],
+    log_scale: bool,
+    numerical: bool,
+) -> _SliceSubplotInfo:
+
+    if target is None:
+
+        def _target(t: FrozenTrial) -> float:
+            return cast(float, t.value)
+
+        target = _target
+
+    return _SliceSubplotInfo(
+        param_name=param,
+        x=[t.params[param] for t in trials if param in t.params],
+        y=[target(t) for t in trials if param in t.params],
+        trial_numbers=[t.number for t in trials if param in t.params],
+        is_log=log_scale,
+        is_numerical=numerical,
+    )
+
+
+def _get_slice_plot_info(
+    study: Study,
+    params: Optional[List[str]] = None,
+    target: Optional[Callable[[FrozenTrial], float]] = None,
+    target_name: str = "Objective Value",
```

Comment:
I fix the codes as follows. Thanks, @himkt -san.
```suggestion
    params: Optional[List[str]],
    target: Optional[Callable[[FrozenTrial], float]],
    target_name: str,
```